## Table of Contents
* [Question 1](#Question1)
* [Question 2](#Question2)
* [Question 3](#Question3)
* [Extended Challenge](#ExtendedChallenge)

In [225]:
import numpy as np
import statistics as stats
import pandas as pd

In [226]:
name = ['Adam', 'Anna', 'James', 'Corey', 'Alisa', 'Jodie', 'John']
math = [11, 12, 11, 7, 9.5, 15, 18]
science = [9, 11, 18, 15.5, 12, 18.5, 15.5]
reading = [16, 8.5, 11.5, 11, 10.5, 7, 11]
history = [13.5, 10, 9, 14, 14, 12, 7.5]

## Question 1: <a class="anchor" id="Question1"></a>

In [228]:
math_mean = np.mean(math)
science_mean = np.mean(science)
reading_mean = np.mean(reading)
history_mean = np.mean(history)
print(f"The means for each subject are:\nMath: {math_mean}\nScience: {science_mean}\nReading: {reading_mean}\nHistory: {history_mean}")

The means for each subject are:
Math: 11.928571428571429
Science: 14.214285714285714
Reading: 10.785714285714286
History: 11.428571428571429


In [229]:
median_math = np.median(math)
print(f"The median math grade is: {median_math}")

The median math grade is: 11.0


In [230]:
mode_history = stats.mode(history)
print(f"Mode for History: {mode_history}")

Mode for History: 14


## Question 2: <a class="anchor" id="Question2"></a>

In [232]:
df_grades = pd.read_csv('student-data.csv')
df_grades[['Math', 'Science', 'Reading', 'History']].corr()

Math   Science   Reading   History
Math     1.000000  0.269494 -0.268562 -0.718358
Science  0.269494  1.000000 -0.527606 -0.358688
Reading -0.268562 -0.527606  1.000000  0.206794
History -0.718358 -0.358688  0.206794  1.000000

### Answer
* Math and history have the strongest correlation coefficient at -0.718

## Question 3: <a class="anchor" id="Question3"></a>

In [235]:
import math
math_grades = [24, 5, 15, 60, 54, 82, 99, 80, 70, 98, 93, 60, 33, 22, 65, 61, 51, 58, 83, 86, 42, 67, 60]
def desc_stats():
    grades_mean = np.mean(math_grades)
    grades_median = np.median(math_grades)
    grades_mode = stats.mode(math_grades)
    grades_min = np.min(math_grades)
    grades_max = np.max(math_grades)
    grades_range = grades_max - grades_min
    grades_var = stats.variance(math_grades)
    grades_std_dev = math.sqrt(grades_var)
    print(f"Mean: {grades_mean}\nMedian: {grades_median}\nMode: {grades_mode}\nMin: {grades_min}\nMax: {grades_max}\nRange: {grades_range}\nVariance: {grades_var}\nStandard Deviation: {grades_std_dev}")

desc_stats()

Mean: 59.47826086956522
Median: 60.0
Mode: 60
Min: 5
Max: 99
Range: 94
Variance: 694.3517786561265
Standard Deviation: 26.350555566365703


## Extended Challenge: <a class="anchor" id="ExtendedChallenge"></a>

* Given the range of scores is quite dramatic and a high standard deviation I would curve the results. Specifically while there are both high and low outliers which screw the median, the both the mode and median are 60, which is an "F" grade.